src/ucp/core/ucp_ep.c

```diff
@@ -2730,7 +2730,9 @@ static void ucp_ep_req_purge(ucp_ep_h ucp_ep, ucp_request_t *req,
         }
 
         ucp_request_put(req);
-    } else if (req->send.uct.func == ucp_rma_sw_proto.progress_get) {
+    } else if ((req->send.uct.func == ucp_rma_sw_proto.progress_get) ||
+               (req->send.uct.func == ucp_amo_sw_proto.progress_fetch) ||
+               (req->send.uct.func == ucp_amo_sw_proto.progress_post)) {
```

Comment:
do we really need it for "ucp_amo_sw_proto.progress_post"?

Reply:
no, we don't need
I've checked again and found that it couldn't be here - https://github.com/openucx/ucx/blob/ec857dc59c804b78572434d8d8e5a6e1ffd48eba/src/ucp/rma/amo_sw.c#L77
fixed